### DISCUSSION TITLE
Is no-std Compatibility A Goal For bevy_hecs?

### DISCUSSION THREAD

**Discussion by zicklag** [bevyengine/bevy#705] [Q&A] [answered]
I wanted to double-check whether or not no-std compatibility was a goal for bevy_hecs or not. I'm trying to find out whether or not I can use `Vec` as part of the implementation of dynamic systems ( #641 ) or if I should feature gate it. A quick search for `use std` shows that it looks like we *are* using non-gated `std` modules in some places in hecs today:

https://github.com/bevyengine/bevy/blob/c32e637384936d973918fb59d1c37c0a4f4f41f7/crates/bevy_ecs/hecs/src/world.rs#L614-L615

Is no-std support a non-goal for `bevy_hecs` or should I still feature-gate/avoid the use of `Vec`?

**Comment by cart** [comment, marked as answer]
Yup I've given up on no_std for bevy_hecs. The rest of bevy has no intention of being no_std, so I didn't really see the point of restricting us in bevy_hecs
